spec/solver_spec.rb

```diff
@@ -0,0 +1,102 @@
+require_relative '../solver'
+
+describe Solver do
+  it 'should return 1 for 0' do
+    expect(Solver.factorial(0)).to eq(1)
+  end
+  it 'should return 1 for 1' do
+    expect(Solver.factorial(1)).to eq(1)
+  end
+  it 'should return 2 for 2' do
+    expect(Solver.factorial(2)).to eq(2)
+  end
+  it 'should return 6 for 3' do
+    expect(Solver.factorial(3)).to eq(6)
+  end
+  it 'should return 24 for 4' do
+    expect(Solver.factorial(4)).to eq(24)
+  end
+  it 'should return 120 for 5' do
+    expect(Solver.factorial(5)).to eq(120)
+  end
```

Comment:
- [ ] The #factorial method need to have tests that handle negative value which raises an exception as an expectation
![image](https://user-images.githubusercontent.com/62705947/194334723-08faf5ee-e46d-4ca9-826b-49ec5990cbfd.png)